src/FSharp.AWS.DynamoDB/TableContext.fs

```diff
@@ -114,17 +115,22 @@ module internal UpdateTableRequest =
             apply tc sc request
             if customize request then Some request else None
 
-    let execute (client: IAmazonDynamoDB) request : Async<TableDescription> = async {
+    let execute (client: IAmazonDynamoDB) (request: UpdateTableRequest) : Async<TableDescription> = async {
         let! ct = Async.CancellationToken
-        let! response = client.UpdateTableAsync(request, ct) |> Async.AwaitTaskCorrect
+        use activity = Diagnostics.startTableActivity request.TableName "UpdateTable" []
+        let! response = client.UpdateTableAsync(request, ct) |> Task.addActivityException activity |> Async.AwaitTaskCorrect
```

Comment:
awaittaskcorrect will do similar unwrapping to what your helper does?

Reply:
Most applications will be adding the exception to the parent span as well, but it would usually be added to the span where it originated - the application may catch & do something else, but that doesn’t change that this span failed. We certainly want to intercept & set activity status to Error at the very least.

Reply:
fair enough I guess. @nordfjord should Equinox.MessagDb etc be following suit if so?

Reply:
We should be catching exceptions and adding them to the spans in messagedb. But I also have been procrastinating homogenising the tracing between the js and f# versions